colpali_engine/models/idefics3/colidefics3/processing_colidefics3.py

```diff
@@ -27,23 +26,35 @@ def image_token_id(self) -> int:
     def process_images(
         self,
         images: List[Image.Image],
+        contexts_prompts: Optional[List[str]] = None,
     ) -> BatchEncoding:
         """
         Process images for ColIdefics3.
         """
         texts_doc: List[str] = []
         images = [[image.convert("RGB")] for image in images]
 
-        for _ in images:
-            messages_doc = [
-                {
-                    "role": "user",
-                    "content": [
-                        {"type": "text", "text": "Describe the image."},
-                        {"type": "image"},
-                    ],
-                },
-            ]
+        for i, _ in enumerate(images):
```

Comment:
why haha ?
either in range(len(images)) or just assert lengths of context prompts and images are the same, then iterate on contecxt_prompts, or again, zip(list(images, list(context)). No reason to use indixes here